src/hooks/useLogin.ts

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+import { axiosInstance } from "../axiosConfig/axiosInstance";
+import { AuthContext } from "../context/AuthContext";
+import { FormDataLogin } from "../interfaces/interface";
+import { toast } from "react-toastify";
+
+const useLogin = () => {
+  const [loading, setLoading] = React.useState(false);
```

Comment:
It is preferable for boolean values to have is before its name like `isLoading` and `isPasswordShown`. This indicates that it is a boolean value. Also a name like `showPassword` feels like a function that will show the password if it is hidden not a value representing whether a password is shown or not.